xapian-letor/api/featurelist.cc

```diff
@@ -104,12 +106,15 @@ FeatureList::create_feature_vectors(const Xapian::MSet & mset,
 	Xapian::Document doc = i.get_document();
 	std::vector<double> fvals;
 	internal->set_data(letor_query, letor_db, doc);
+	Feature::Internal *internal_feature = new Feature::Internal();
+	// set all the parameters for internal_feature
+	internal_feature->set_database(letor_db);
+	internal_feature->set_query(letor_query);
+	internal_feature->set_doc(doc);
```

Comment:
We don't ever use these set methods except here - we could get rid of them completely and pass these 3 values as parameters to the constructor.